### PR TITLE
Markdown Previewer Iframe Tests Option

### PR DESCRIPTION
**Pull request by jnmorse**
#### Pre-Submission Checklist

- [x] Your pull request targets the `master` branch of freeCodeCamp/testable-projects-fcc
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] Your changes have been tested either locally or using a newly created CDN based on your fork's testable-projects-fcc/build/bundle.js file
- [x] Working changes have been added to the build by running `npm run build`

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [ ] Closes currently open issue: Closes #XXX

#### Description

Added a class that will grab content from an iframe if it exists with
the id preview, and the inner div of a div with id root.

replaced all calls for preview, or document with PreviewDocument
class static method calls.

Also snuck in another change here to the links test that checks that
all links target _blank without the forEach. The way it was currently
designed would pass if there were no links being grabbed and other
tests were failing. Will remove if necessary.

### CODE REVIEW

**Comment by tbushman** [comment]
@jnmorse You should remove the target `_blank` bit, and first create an issue about it, then submit a separate PR for it. Thanks!

File: src/project-tests/markdown-previewer-tests.js [hunk 1]
```diff
@@ -8,15 +8,43 @@ export default function createMarkdownPreviewerTests() {
...
+    
+    /**
+     * Using class PreviewDocument to be a proxy for grabing the document
+     * information. In case this is an iframe, use the contentDocument of
+     * the iframe.
+     */
+    class PreviewDocument {
+      static get document() {
+        if (preview && preview.contentDocument) {
+          return preview.contentDocument.querySelector('div#root div');
```
Comment (tbushman) [inline src/project-tests/markdown-previewer-tests.js]: Will this be a customary arrangement for campers choosing to use an iframe? Will we need to introduce a test for the existence of the `div#root div` element?

**Comment by tchaffee** [comment]
@jnmorse I'm not convinced we need this flexibility or this change. Using an iframe isn't part of the requirements, is it? Is it included in the lessons around this project? If it comes up as an issue, I would just tell campers to not use an iframe. We have to be careful about trying to support every possible solution and instead make sure the tests guide the campers step-by-step to one solution. Am I missing something?

**Comment by jnmorse** [comment]
That is fine, to be honest, I'd need to tweak this some anyways. Ended up noticing it breaks test from running less it works, Though I'll have to change my solution from using an iframe to just outputting to a div. I'll close this for now. Far as I know I'm the only one who chose to use an iframe so far anyway.

**Comment by tchaffee** [comment]
Thanks. I appreciate your understanding on this. For this phase of the project we've got to be pretty focused on making the tests specific, and to point to one solution. Because the next step will to be to try to get the tests to work on more browsers. And then to support more frameworks (Vue, Angular). If we did all that AND tried to support multiple approaches to a project, I don't think we would ever be done fixing bugs. I hope the overall strategy makes sense.

**Pull request closed without merging**